Sources/Keychain/KeychainHelper.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  File.swift
+//  ETDistribution
+//
+//  Created by Itay Brenner on 30/10/24.
+//
+
+import Foundation
+
+enum KeychainError: LocalizedError {
+  case unexpectedStatus(OSStatus)
+}
+
+enum KeychainHelper {
+  static let service = "com.emerge.ETDistribution"
+  
+  static func setToken(_ token: String, key: String) throws {
+    let existingToken = try getToken(key: key)
+    if existingToken == nil {
+      try addToken(token, key: key)
+    } else {
+      try updateToken(token, key: key)
+    }
+  }
+  
+  static func getToken(key: String) -> String? {
+    let query = [
+        kSecClass: kSecClassGenericPassword,
+        kSecAttrService: service,
+        kSecAttrAccount: key,
+        kSecMatchLimit: kSecMatchLimitOne,
+        kSecReturnData: true
+    ] as CFDictionary
+
+    var result: AnyObject?
+    let status = SecItemCopyMatching(query, &result)
```

Comment:
Apple says to call these functions on a background thread: https://developer.apple.com/documentation/security/secitemcopymatching(_:_:)#Performance-considerations